File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";
    int idx = 0; // 단어 내의 인덱스를 관리

    for (char c : s) {
        if (c == ' ') { // 공백 처리
            answer += c;
            idx = 0; // 공백이 나오면 단어 인덱스를 초기화
        } else {
            if (idx % 2 == 0) {
                answer += toupper(c); // 짝수 인덱스 대문자
            } else {
                answer += tolower(c); // 홀수 인덱스 소문자
            }
            idx++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        StringBuilder answer = new StringBuilder();
        int idx = 0;

        for (char c : s.toCharArray()) {
            if (c == ' ') {
                answer.append(c);
                idx = 0;
            } else {
                answer.append(idx % 2 == 0 ? Character.toUpperCase(c) : Character.toLowerCase(c));
                idx++;
            }
        }

        return answer.toString();
    }
}